cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfArgumentsCreator.java

```diff
@@ -0,0 +1,94 @@
+package cloud.foundry.cli.crosscutting.mapping;
+
+import picocli.CommandLine;
+import picocli.CommandLine.ParseResult;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Properties;
+
+import cloud.foundry.cli.crosscutting.logging.Log;
+
+/**
+ * This utility class creates certain options with default values, which were not passed when the command was called.
+ * There are currently default values for <code>api</code>, <code>organization</code> and <code>space</code>.
+ */
+public class CfArgumentsCreator {
+
+    /**
+     * Name of the property file, which contains default values
+     */
+    private static final String CF_CONTROL_PROPERTIES = "cf_control.properties";
+
+    /**
+     * The method determines the commandline arguments.
+     * The following specification applies:
+     * when the command is called and one of the option <code>space</code> or / and <code>api</code>
+     * or / and <code>organization</code> is missing,
+     * the method extends the CommandLine with the corresponding option and
+     * sets a default value for this, which has been configured in a property file.
+     * <p>
+     * For example, it converts the command
+     * "command -s space value -a api value"
+     * into
+     * "command -s spaceValue -a apiValue -o DefaultValueForOrganization"
+     * by appending the -o with its default value.
+     *
+     * @param cli  CommandLine interpreter
+     * @param args Commandline arguments
+     * @return Commandline arguments
+     */
+    public static String[] determineCommandLine(CommandLine cli, String[] args) {
+        List<String> optionNames = Arrays.asList("-a", "-o", "-s");
+        List<String> missingOptions = new ArrayList<>();
+
+        ParseResult parseResult = cli.parseArgs(args);
+        while (parseResult.hasSubcommand()) {
```

Comment:
Please document the purpose of this while loop. I'm not entirely sure why you need to recurse down to the lowest of all subcommands.